tap_appstore/client.py

```diff
@@ -18,133 +23,117 @@
 
 _Auth = Callable[[requests.PreparedRequest], requests.PreparedRequest]
 
-# TODO: Delete this is if not using json files for schema definition
-SCHEMAS_DIR = importlib_resources.files(__package__) / "schemas"
-
+logger = logging.getLogger(__name__)
 
-class AppStoreStream(RESTStream):
+class AppStoreStream(Stream):
     """AppStore stream class."""
+    DATE_FORMAT = '%Y-%m-%d'
+    DATE_INCREMENT = timedelta(days=1)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.api = self.setup_api_connection()
+        self.skip_line_first_values = []
+
+        date_format_mapping = {
+            'begin_date': '%m/%d/%Y',
+            'start_date': '%m/%d/%Y',
+            'end_date': '%m/%d/%Y',
+            '_api_report_date': '%Y-%m-%d',
+        }
```

Comment:
Let's move this date format to the downstream class keeping only the fields of each class.